arrows/core/render_mesh_depth_map.cxx

```diff
@@ -0,0 +1,213 @@
+ * \brief Header for core render_mesh_depth_map function
+ */
+
+#include "render_mesh_depth_map.h"
+
+#include <vital/types/camera_perspective.h>
+#include <vital/types/image.h>
+#include <vital/types/vector.h>
+
+
+namespace {
+
+/// Helper function to check if a point is inside a triangle
+bool is_point_inside_triangle(const kwiver::vital::vector_2d& p,
+                              const kwiver::vital::vector_2d& a,
+                              const kwiver::vital::vector_2d& b,
+                              const kwiver::vital::vector_2d& c)
+{
+  kwiver::vital::vector_2d AB = b - a;
+  kwiver::vital::vector_2d AC = c - a;
+  kwiver::vital::vector_2d AP = p - a;
+  double inv_total_area = 1.0 / (AB[0] * AC[1] - AB[1] * AC[0]);
+  double area_1 = inv_total_area * (AB[0] * AP[1] - AB[1] * AP[0]);
+  double area_2 = inv_total_area * (AP[0] * AC[1] - AP[1] * AC[0]);
+  return area_1 >= 0 && area_2 >= 0 && (area_1 + area_2) <= 1;
+}
+
+/// Helper function to compute barycentric coordinates of p w.r.t the triangle abc
+kwiver::vital::vector_3d barycentric_coordinates(const kwiver::vital::vector_2d& p,
+                                                 const kwiver::vital::vector_2d& a,
+                                                 const kwiver::vital::vector_2d& b,
+                                                 const kwiver::vital::vector_2d& c)
+{
+  kwiver::vital::matrix_3x3d abc;
+  abc << a, b, c, 1, 1, 1;
+  double det_inv = 1.0 / abc.determinant();
+  kwiver::vital::vector_3d res;
+  res(0) = ((b(1) - c(1)) * (p(0) - c(0)) - (b(0) - c(0)) * (p(1) - c(1))) * det_inv;
+  res(1) = ((c(1) - a(1)) * (p(0) - c(0)) - (c(0) - a(0)) * (p(1) - c(1))) * det_inv;
+  res(2) = 1.0 - res(0) - res(1);
+  return res;
+}
+
+}
+
+namespace kwiver {
+namespace arrows {
+
+
+vital::image_container_sptr render_mesh_depth_map(vital::mesh_sptr mesh, vital::camera_perspective_sptr camera)
+{
+  unsigned int nb_vertices = mesh->num_verts();
+  vital::mesh_vertex_array<3>& vertices = dynamic_cast< vital::mesh_vertex_array<3>& >(mesh->vertices());
+
+  // Project all points on image
+  std::vector<vital::vector_2d> points_2d(nb_vertices);
+  for (unsigned int i = 0; i < vertices.size(); ++i)
+  {
+    points_2d[i] = camera->project(vertices[i]);
+  }
+
+  // Compute the points negative inverse depth
+  std::vector<double> points_depth(nb_vertices);
+  for (unsigned int i=0; i < vertices.size(); ++i)
+  {
+    points_depth[i] = -1.0 / camera->depth(vertices[i]);
+  }
+
+  // Initialize z_buffer with max infinity
+  vital::image_of<double> z_buffer(camera->image_width(), camera->image_height(), 1);
+  for (size_t j = 0; j < z_buffer.height(); ++j)
+  {
+    for (size_t i = 0; i < z_buffer.width(); ++i)
+    {
+      z_buffer(i, j) = std::numeric_limits<double>::infinity();
+    }
+  }
+
+  if (mesh->faces().regularity() == 3)
+  {
+    rasterize_depth(dynamic_cast< const vital::mesh_regular_face_array<3>& >(mesh->faces()), points_2d, points_depth, z_buffer);
```

Comment:
You might consider capturing the cast face array into a variable before passing into the function.
```c++
auto const& triangles = dynamic_cast< vital::mesh_regular_face_array<3>const& >(mesh->faces());
rasterize_depth(triangles, points_2d, points_depth, z_buffer);
```

It might help with the dashboard failures and it's easier to read.

Reply:
I tried but it doesn't solve the problem.